src/metabase/driver/sql_jdbc/actions.clj

```diff
@@ -303,3 +303,57 @@
             [(conj errors {:index row-index, :error (ex-message e)})
              created-rows]))))
      rows)))
+
+(defn- pk-values->filters [table-id pk-values]
+  ;; check that all pk-values have the same shape
+  (when-not (apply = (map (comp set keys) pk-values))
+    (throw (ex-info (trs "Inconsistent row selection.")
```

Comment:
Might be worth making this error message a little bit more detailed. If was scratching my head when I first saw this error, and it was only by reading the comment above that I was able to figure out what the error meant. Maybe something like `The specified rows have different sets of keys: found #{:id}, #{:name}` would make this a little clearer. You can make that like of `#{:id}, #{:name}` programmatically